tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Requests;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Responses;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps.Common;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Support;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Utils;
+using NHSD.BuyingCatalogue.Ordering.Api.Testing.Data.Entities;
+using TechTalk.SpecFlow;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps
+{
+    [Binding]
+    internal sealed class OrderItemDeleteSteps
+    {
+        private readonly Request request;
+        private readonly Response response;
+        private readonly Settings settings;
+        private DeleteOrderItemRequest deleteOrderItemRequest;
+        private int originalOrderItemsCount;
+
+        public OrderItemDeleteSteps(
+            Request request,
+            Response response,
+            Settings settings)
+        {
+            this.request = request;
+            this.response = response;
+            this.settings = settings ?? throw new ArgumentNullException(nameof(settings));
+        }
+
+        [Given(@"the user creates a request to delete order item with catalogue item ID (.*) for order ID (\d{1,6})")]
+        public async Task GivenTheUserCreatesARequestToDeleteOrderItemForOrder(string catalogueItemId, int orderId)
+        {
+            try
+            {
+                originalOrderItemsCount = await OrderItemEntity.GetItemIdsForOrderId(
+                        settings.ConnectionString,
+                        orderId);
+            }
+            catch (SqlException)
+            {
+            }
```

Comment:
The SQL call before test is required to avoid hardcoding the figure. 
That fails for the service failure case and doesn't let the rest of case execute.
I can hard code the number of order items but that would need updating every time order items are changed.
Lemme know if you'd like the code changed here.

Reply:
That makes sense, thanks. However, I think we have an alternative option.

We can take a dependency of `OrderContext`. The `order items exist` step will add each item to the `OrderContext.OrderItemReferenceList` dictionary (keyed by order ID), so it should be possible to get a count of order items without hard-coding it or going to the database. Might this approach work?

Reply:
Nice solution, thanks, done.